Storage/src/Connection/Rest.php

```diff
@@ -614,14 +622,22 @@ private function buildDownloadObjectParams(array $args)
             'restDelayFunction' => null
         ]);
 
+        $queryOptions = [
+            'generation' => $args['generation'],
```

Comment:
Please add default values for `generation` and `userProject`. Else, it may throw key not found exception.

Reply:
Oh okay, I thought these are new parameters. Ignore this comment.